### PR TITLE
adding evaluation code

### LINKED ISSUE #25: Model Evaluation After Successful Training
The CodeT5 model has been successfully trained using the bioinformatics R code dataset. The next step is to evaluate the model’s performance using appropriate metrics and benchmarks to assess how well it generalizes to unseen bioinformatics-related coding tasks.

Acceptance Criteria:

-  Define evaluation tasks and datasets 
-  Evaluate model using relevant metrics 
-  Document qualitative results
-  Identify areas for improvement or potential follow-up training

### PR DESCRIPTION
**Pull request by PranaviKolouju**
Fixes #25 

**What was changed?**

I created a python script that was used to test the fine-tuned CodeT5 model's ability to generate functional R code for basic DNA/RNA operations. It was also used to compare evaluation metrics against the base CodeT5 model. 

**Why was it changed?**

In order for us to assess whether our finetuning efforts were successful, we need an evaluation framework that allows us to compare the base CodeT5 model to the fine-tuned CodeT5 model. This code presents a script that tests the accuracy of generated code against defined test cases, allowing us to measure whether the fine-tuned model is generating sensible/functioning code. 

**How was it changed?**

I created a script `eval.py` that contains 16 test cases, prompting the fine-tuned model to create specific R-based functions that perform basic DNA/RNA operations, such as counting the number of nucleotides in a DNA sequence or translating a DNA sequence to a sequence of amino acids. The code generated from the model is loaded into an .R file and is run to see if it passes the written test case. I evaluate on the Pass@K metric which measure the frequency at which the correct output is generated among K samples. I used K = 1, 5, 10, and 20 similar to the BioCoder benchmarking framework. Below are the evaluation metrics for both the base CodeT5 model and the ffine-tuned CodeT5 model, showing how the finetuning was unsuccessful. 

image

### CODE REVIEW

**Review by AjithAkuthota23 — APPROVED** [review]
Good work
